src/harness/harness.ts

```diff
@@ -133,17 +132,16 @@ namespace Utils {
         return content;
     }
 
-    export function memoize<T extends Function>(f: T): T {
+    export function memoize<T extends Function>(f: T, memoKey?: (...anything: any[]) => string): T {

```

Comment:
I would just make `memoKey` non-optional and use `path => path` for `directoryName`. Also comment that its arguments are the same as the arguments to `f`.